internal/search/query/parser.go

```diff
@@ -216,6 +216,68 @@ func ScanParameter(parameter []byte) Parameter {
 	return Parameter{Field: "", Value: string(parameter)}
 }
 
+// ScanSearchPattern scans for a pattern using a heuristic that allows it to
+// contain parentheses, if balanced, with appropriate lexical handling for
+// traditional escape sequences, escaped parentheses, and escaped whitespace.
+func ScanSearchPatternHeuristic(buf []byte) ([]string, int, bool) {
+	var count, advance, balanced int
+	var r rune
+	piece := []rune{}
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

```suggestion
	var piece []rune
```